src/ec/eddsa.rs

```diff
@@ -133,33 +174,144 @@ impl signature::VerificationAlgorithm for EdDSAParameters {
     fn verify(&self, public_key: untrusted::Input, msg: untrusted::Input,
               signature: untrusted::Input) -> Result<(), error::Unspecified> {
         let public_key = public_key.as_slice_less_safe();
-        if public_key.len() != 32 || signature.len() != 64 {
+        if public_key.len() != PUBLIC_KEY_LEN || signature.len() != SIGNATURE_LEN {
             return Err(error::Unspecified);
         }
         let msg = msg.as_slice_less_safe();
         let signature = signature.as_slice_less_safe();
-        bssl::map_result(unsafe {
-            GFp_ed25519_verify(msg.as_ptr(), msg.len(), signature.as_ptr(),
-                               public_key.as_ptr())
-        })
+
+        let mut a = ExtPoint::new_at_infinity();
+        if (signature[63] & 224) != 0 {
+            return Err(error::Unspecified);
+        }
+        try!(bssl::map_result(unsafe {
+            GFp_x25519_ge_frombytes_vartime(&mut a, slice_as_scalar_ref(public_key).unwrap())
+        }));
+        for i in 0..ELEM_LIMBS {
+            a.x[i] = -a.x[i];
+            a.t[i] = -a.t[i];
+        }
+        let mut r_copy = [0u8; SCALAR_LEN];
+        r_copy.copy_from_slice(&signature[..SCALAR_LEN]);
+        let mut s_copy = [0u8; SCALAR_LEN];
+        s_copy.copy_from_slice(&signature[SCALAR_LEN..]);
+        let mut h = [0u8; digest::SHA512_OUTPUT_LEN];
+        h.copy_from_slice({
+            let mut ctx = digest::Context::new(&digest::SHA512);
+            ctx.update(&signature[..SCALAR_LEN]);
+            ctx.update(&public_key);
+            ctx.update(&msg);
+            ctx.finish().as_ref()
+        });
+        unsafe {
+            GFp_x25519_sc_reduce(slice_as_scalar_ref_mut(&mut h).unwrap())
+        };
+        let mut r = Point::new_at_infinity();
+        unsafe {
+            GFp_ge_double_scalarmult_vartime(&mut r,
+                                             slice_as_scalar_ref(&h).unwrap(),
+                                             &a,
+                                             slice_as_scalar_ref(&s_copy).unwrap())
+        };
+        let mut r_check = [0u8; ELEM_LEN];
+        unsafe { GFp_x25519_ge_tobytes(&mut r_check, &r) };
+        constant_time::verify_slices_are_equal(&r_check, &r_copy)
```

Comment:
I think you don't need to benchmark it. Just switch to the normal slice comparison. I just noticed today we're already using the normal slice comparison in `EdDSAKeyPair::from_bytes()`.